---
title: "Report Open Data Analysis -  Bidirectional Best Hit (BBH) analysis"
author: "Ericka M. Hernandez Benitez"
date: "`r Sys.Date()`"
output: 
  html_document:
    df_print: paged
    code_folding: hide
    self_contained: yes
    fig_caption: yes
    highlight: zenburn
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
bibliography: ./references.bib
---

## Theoretical basis of the project 

<div align="justify">

To assemble a Transcriptional Regulatory Network (TRN) for *Rhizobium phaseoli*  we can take advantage of orthology and the well-characterized *E. coli* TRN [@santoszavaleta_2019]. Orthologous genes are derived from a single ancestral gene in the last common ancestor of the compared species [@koonin_2005]. Organisms that are phylogenetically close, tend to have a smaller variation between genes and thus, approaches involving sequence alignments are performed in order to predict orthologous genes [@lin_2006]. Bidirectional best hit (BBH) is a common method to infer orthology, it identifies pairs of genes in two different genomes that are more similar to each other than either is to any other gene [@dalquen_2013]. Gene pairs with the highest levels of sequence similarity are labeled as orthologous [@lin_2006]. To compute sequence similarity  one of the most common bioinformatics tools is the Basic Local Alignment Search Tool (BLAST) [@altschul_1990] which can compare either nucleotide (BLASTn) or protein sequences (BLASTp) against sequence databases. However, selecting the appropriate BLAST program (e.g. BLASTn, BLASTp, BLASTx) and the parameter’s thresholds are factors that should be carefully chosen [@dalquen_2013], since there are often no simple reciprocal relationships between genes due to gene duplication, deletion, and inversion [@lin_2006]. 

The orthology method takes advantage of the knowledge and discoveries available in model  species (e.g *E. coli*). It is based on the premise that TF-Target Gene (TG) relationships proven in one organism can be conserved in another one. To establish the functional counterpart for any TF-TG pair, and thus reconstruct a TRN, it is necessary to identify the presence of orthologous genes in different species. The method assumes that not only the DNA sequence is conserved, but also the gene function. Additional analysis (e.g DNA-binding motif search, and gene expression data analysis, among others) are required to to corroborate the function conservation [@mercatelli_2020]. 

</div>
\nbsp

### Statement of the problem

<div align="justify">

Here we have data from a BBH using the protein sequences of *E. coli* and *R. phaseoli*. In order to add information regarding the functional conservation we are going to perform an analysis motif. Particularly, we are going to focus on the functional conservation of TFs (which are proteins that bing to DNA sequence and activate/repress the transcription of their regulated genes). We expected that if the orthologous TFs identified by BBH have a functional conservation the sequences that aligned in the BBH must be enriched with DNA-binding motifs. 

</div>

## Question to be addressed

<div align="justify">

\nbsp

Do BBH output sequences have DNA-binding motifs?

\nbsp

If so, what is the proportion of DNA-binding motifs in comparison to the other type of motifs found in the BBH sequences?

</div>
\nbsp

## Used datasets

\nbsp

We account with **five** raw datasets:

\nbsp

<div align="justify">

1. **ecoliAnnotation.tsv**: This file contains the relationships between protein identifier, gene name (according to different databases), locus tag, synonyms of the genes' name, and the product) from *E. coli K-12* genome. 
This file was generated by the intersection of the following databases: Ecocyc [@karp_2018], Regulondb [@santoszavaleta_2019], Abasy Atlas [@escorciarodrguez_2020] and NCBI genome [@kuznetsov_2021].

</div>

```{r}
ecoliAnnotation   <- read.table(file = "../data/raw/ecoliAnnotation.tsv", header = TRUE , sep = "\t")
head(ecoliAnnotation)
```

\nbsp
\nbsp

<div align="justify">

2. **geneAASeq.tsv**: This file contains the aminoacid sequences of  *E. coli K-12* genome. It was retrieved from NCBI genome [@kuznetsov_2021].

</div>

```{r}
geneAASeq <- read.table(file = "../data/raw/geneAASeq.tsv", header = FALSE , sep = "\t")
colnames(geneAASeq) <- c("ProteinID", "Sequence")
head(geneAASeq)
```

\nbsp
\nbsp

<div align="justify">

3. **MotifsSeqRelation.tsv**: This file contains the relationships between genes and its annotated motifs (both description and coordinates in the protein) from *E. coli K-12* genome. This file was retrieved from Ecocyc [@karp_2018]. 

</div>
```{r}
motifsSeqRelation <- read.table(file = "../data/raw/MotifsSeqRelation.tsv", header = FALSE , sep = "\t")
colnames(motifsSeqRelation) <- c("TF_name", "Locus_tag", "Motif_description", "mSS", "mSE")
head(motifsSeqRelation)
```

\nbsp
\nbsp

<div align="justify">

4. **Orthologous_ECaaq_RZaadb_blastN_b1_m8.tab**: This file contains the results of a BBH performed with BLASTp [@altschul_1990].
</div>

```{r}
blastResult <- read.table(file = "../data/raw/Orthologous_ECaaq_RZaadb_blastN_b1_m8.tab", header = TRUE , sep = "\t")
head(blastResult)
```

\nbsp
\nbsp

<div align="justify">
5. **TFs_coli.txt**: This file contains a list of TF in **E. coli K-12**. It was retrieved from Regulondb [@santoszavaleta_2019]. 
</div>

```{r}
tfsEcoli <- read.table(file = "../data/raw/TFs_coli.txt", header = FALSE , sep = "\t")
colnames(tfsEcoli)[1] <- "TF_name"
head(tfsEcoli)
```

\nbsp

## Methods and modular workflow

<div align="justify">

The workflow consist of three main parts (Figure 1) each one associated with an script in R.

![**Figure 1.** Workflow. First, we cleaned the data from different sources and merged them into one single table. Then, we extracted the sequences from both BBH output and motifs. Finally, we asked whether the DNA-binding motifs aligned with the BBH output.](../figs/framework.png)

</div>
\nbsp




### Motif analysis
<div align="justify">

We retrieved the following Ecocyc [@karp_2018] annotated motifs: Ca-Binding-Region, Conserved-Region, Catalytic-Domain, DNA-Binding-Region, Intramembrane-Region, Nucleotide-Phosphate-Binding-Region, Protein-Structure-Region, Alpha-Helix-Region, Beta-Strand-Region, Coiled-Coil-Region, Transmembrane-Region, and Zn-Finger-Region. For each of the orthologus TF, we retrieved the available annotated motifs and observed if they were present in the sequence that was aligned in the BBH analysis.

</div>

\nbsp


## Results

<div align="justify">

Here, we defined three possible scenarios when searching for an Ecocyc annotated motif: ‘all’, ‘at least one’, or ‘none’ of the motifs described in Ecocyc could be present in the BLASTp output sequences. We considered a ‘none’ subcategory (‘none NAs’) when a particular TF did not account for a single motif reported in Ecocyc. Additionally, we examined the types of motifs (e.g DNA-Binding-Region, Conserved-Region, Transmembrane-Region, among others) found in each category

\nbsp
 
We found that ~42%, ~34%, and ~5% of the oTFs had ‘all’, ‘at least one’, and ‘none’ of the Ecocyc annotated motifs present in the BLASTp output sequences, respectively.  Around 17% of oTFs did not have a single annotated motif in Ecocyc (**Figure 2A**). The majority of motif types found in the ‘all’ category were conserved regions (48%) and DNA-binding regions (44%) (**Figure 2B**), whereas, in the ‘at least one’ category the predominant type of found motifs was Conserved Regions (~44%) (**Figure 2C**).

</div>

<div align="justify">
![**Figure 9.** Motif analysis. We analyzed the presence and proportion of several Ecocyc-annotated motifs in the orthologous TFs. A) Shows the proportion of oTFs having ‘all’, ‘at least one’ and ‘none’ of the Ecocyc-annotated motifs. The category ‘None motif (0's)’ describes TFs motifs annotated in Ecocyc but not found in the BLAST aligned sequence while the category ‘None motif (NAs)’ represents TFs lacking any kind of motif reported in Ecocyc. In addition, motif classes found in the "all" B) and "at least one" C) and "None motif (0's)" categories are shown.](../figs/pieplotMotifs.png)
</div>

## Requirements in terms of the OS and R libraries


The workflow is implemented in R (**R $\ge$ 4.2.2**) and have dependencies of R packages, Bioconductor packages (**Table 1**).

\nbsp
\nbsp


| OS            | R packages                                          | Bioconductor packages |
|---------------|-----------------------------------------------------|-----------------------|
| Windows/linux | dplyr, ggplot2,ggpubr, RColorBrewer, stringr, tidyr | Biostrings            |

**Table 1.** Dependencies of the workflow.


## References

